packages/camera/camera_avfoundation/lib/src/avfoundation_camera.dart

```diff
@@ -248,14 +248,22 @@ class AVFoundationCamera extends CameraPlatform {
 
   @override
   Future<void> startVideoRecording(int cameraId,
-      {Duration? maxVideoDuration}) async {
+      {Duration? maxVideoDuration,
+      Function(CameraImageData image)? streamCallback,
+      CameraImageStreamOptions? streamOptions}) async {
     await _channel.invokeMethod<void>(
       'startVideoRecording',
       <String, dynamic>{
         'cameraId': cameraId,
         'maxVideoDuration': maxVideoDuration?.inMilliseconds,
+        'enableStream': streamCallback != null,
       },
     );
+
+    if (streamCallback != null) {
+      _installStreamController().stream.listen(streamCallback);
```

Comment:
This is doing 2 complex things in 1 line, making it a bit hard to understand. Can you split it?

Reply:
No problem, done.